### PR TITLE
make filters use security manager

### PR DESCRIPTION
**Pull request by timifasubaa**
This PR removes the redundant has_all_datasource_access method and uses the security manager's all_datasource_access method. It also makes superset filters configurable via the security manager. 

Longer term, it will be nicer to make the whole method configurable so different deployments can decide their own mechanisms to filter the dashboards. 

@john-bodley  @michellethomas  @mistercrunch

### CODE REVIEW

File: superset/security.py [hunk 1]
```diff
@@ -85,6 +85,15 @@ def get_schema_perm(self, database, schema):
         if schema:
             return '[{}].[{}]'.format(database, schema)
 
+    def always_list_all_dashboards(self):
```
Comment (john-bodley) [inline superset/security.py]: I agree with your comment that we should defer to the security manager to answer said question, however I’m not sure we need these always_ methods.

File: superset/views/base.py [hunk 2]
```diff
@@ -248,15 +248,11 @@ def get_view_menus(self, permission_name):
                 vm.add(vm_name)
         return vm
 
-    def has_all_datasource_access(self):
-        return (
-            self.has_role(['Admin', 'Alpha']) or
```
Comment (john-bodley) [inline superset/views/base.py]: Does removing these role checks potentially break things?

Comment (timifasubaa) [reply superset/views/base.py]: There are only three usage of the has_all_datasource_access and they are the ones I changed.

Comment (timifasubaa) [reply superset/views/base.py]: Here is the link to where all_datasource_access is granted to Alpha. 
Note that Admin is granted all the possible priviledges, which includes those contained in Alpha. 
Also linked is the line in the security manager that checks the user has all_datasource_access

https://github.com/apache/incubator-superset/blob/4bf69a7260f1d4c8fbfe9d9c0300cae39ca3bf81/superset/security.py#L71

https://github.com/apache/incubator-superset/blob/4bf69a7260f1d4c8fbfe9d9c0300cae39ca3bf81/superset/security.py#L96

**Review by john-bodley — APPROVED** [review]

**Pull request merged**